### PR TITLE
Added some lambdas for TableView datasource

### PR DESCRIPTION
**Pull request by sergdort**
`commitEditingStyle ` , `moveRowAtIndexPath`,  `section…IndexTitles`, `sectionForSectionIndexTitle`

### CODE REVIEW

File: Sources/DataSources/TableViewSectionedDataSource.swift [hunk 1]
```diff
@@ -128,7 +160,13 @@ public class RxTableViewSectionedDataSource<S: SectionModelType>
     public var titleForFooterInSection: ((section: Int) -> String?)?
     
     public var canEditRowAtIndexPath: ((indexPath: NSIndexPath) -> Bool)?
+    public var commitEditingStyle: ((editingStyle: UITableViewCellEditingStyle, indexPath: NSIndexPath) -> Void)?
+    
     public var canMoveRowAtIndexPath: ((indexPath: NSIndexPath) -> Bool)?
+    public var moveRowAtIndexPath: ((sourceIndexPath: NSIndexPath, destinationIndexPath: NSIndexPath) -> Void)?
+    
+    public var sectionIndexTitles: (() -> [String]?)?
```
Comment (sergdort) [inline Sources/DataSources/TableViewSectionedDataSource.swift]: Not sure if it should be lambda or just `var sectionIndexTitles :  [String]?`

File: Sources/DataSources/TableViewSectionedDataSource.swift [hunk 2]
```diff
@@ -172,9 +210,28 @@ public class RxTableViewSectionedDataSource<S: SectionModelType>
...
+        return commitEditingStyle?(editingStyle: editingStyle, indexPath: indexPath) ??
+            super._tableView(tableView, commitEditingStyle: editingStyle, forRowAtIndexPath: indexPath)
+    }
+    
     override func _tableView(tableView: UITableView, canMoveRowAtIndexPath indexPath: NSIndexPath) -> Bool {
         return canMoveRowAtIndexPath?(indexPath: indexPath) ??
             super._tableView(tableView, canMoveRowAtIndexPath: indexPath)
     }
     
+    override func _tableView(tableView: UITableView, moveRowAtIndexPath sourceIndexPath: NSIndexPath, toIndexPath destinationIndexPath: NSIndexPath) {
```
Comment (kzaher) [inline Sources/DataSources/TableViewSectionedDataSource.swift]: I don't think we need `moveRowAtIndexPath`, we have 

```
public var rx_itemMoved: ControlEvent<ItemMovedEvent> {
```

File: Sources/DataSources/TableViewSectionedDataSource.swift (hunk 1 above)
Comment (kzaher) [inline Sources/DataSources/TableViewSectionedDataSource.swift]: Lambda :+1:

I think we could probably pass `self` as first argument to all lambdas so we don't have to do weak/strong/unowned dance all the time.

Thoughts?

File: Sources/DataSources/TableViewSectionedDataSource.swift (hunk 2 above)
Comment (sergdort) [inline Sources/DataSources/TableViewSectionedDataSource.swift]: Oh, yeh

Completely forgot about it :)

File: Sources/DataSources/TableViewSectionedDataSource.swift (hunk 1 above)
Comment (sergdort) [inline Sources/DataSources/TableViewSectionedDataSource.swift]: Hm.. This is very good point!

**Comment by sergdort** [comment]
Hi, @kzaher , I've made changes and moved them into one commit

**Pull request merged**